evcouplings/couplings/model.py

```diff
@@ -397,6 +387,30 @@ def __read_plmc_v2(self, filename, precision):
                     )
                     self.J_ij[j, i] = self.J_ij[i, j].T
 
+            # if lambda_h is negative, the model was
+            # inferred using mean-field
+            if self.lambda_h < 0:
+                # cast model to mean field model
+                from evcouplings.couplings.mean_field import MeanFieldCouplingsModel
+                self.__class__ = MeanFieldCouplingsModel
+
+                # pseudo-count is saved in lambda_h
+                # (with a negative sign)
+                self.pseudo_count = -self.lambda_h
+                self.lambda_h = None
```

Comment:
If we leave lambda_J and lambda_h as they are instead of setting to None, this would allow to save the model to file again later on without any other changes... but probably the cleaner solution would be to explicitly re-encode as -1 in the to_file method, which would save users from misinterpreting the lambda_h and lambda_J values when using the object...